dummyserver/server.py

```diff
@@ -16,6 +16,11 @@
 import warnings
 from datetime import datetime
 
+try:
+    from ssl import CERT_OPTIONAL
+except:
+    CERT_OPTIONAL = None
```

Comment:
To be clear, I'm asking because if we do blindly `import ssl` into the dummyserver when running tests on an python install that doesn't have the `ssl` module, none of the tests will work, full stop... whereas they might work today, without the import.

On the other hand, keeping the dummyserver free of urllib code, so that we are only testing the urllib code as a client, not having any of it in the dummyserver, makes a certain amount of sense as well.

So which option is less-bad?

I must admit that I'm now curious how many of the tests would fail if run against a python without the `ssl` module...